packages/viewer/src/components/viewer-transform-widget/viewer-transform-widget.tsx

```diff
@@ -0,0 +1,356 @@
+import { Component, h, Host, Prop, Watch } from '@stencil/core';
+import { Point, Vector3 } from '@vertexvis/geometry';
+import { Color, Disposable } from '@vertexvis/utils';
+import classNames from 'classnames';
+
+import { readDOM, writeDOM } from '../../lib/stencil';
+import { TransformController } from '../../lib/transforms/controller';
+import { Mesh } from '../../lib/transforms/mesh';
+import {
+  computeUpdatedPosition,
+  convertCanvasPointToWorld,
+  convertPointToCanvas,
+} from './util';
+import { TransformWidget } from './widget';
+
+@Component({
+  tag: 'vertex-viewer-transform-widget',
+  styleUrl: 'viewer-transform-widget.css',
+  shadow: true,
+})
+export class ViewerTransformWidget {
+  /**
+   * The viewer to connect to measurements. If nested within a <vertex-viewer>,
+   * this property will be populated automatically.
+   */
+  @Prop()
+  public viewer?: HTMLVertexViewerElement;
+
+  /**
+   * The starting position of this transform widget. This position will be updated
+   * as transforms occur. Setting this value to `undefined` will remove the widget.
+   */
+  @Prop({ mutable: true })
+  public position?: Vector3.Vector3;
+
+  /**
+   * The controller that is responsible for performing transforms.
+   */
+  @Prop({ mutable: true })
+  public controller?: TransformController;
+
+  /**
+   * The color of the translation arrow on the x-axis. Defaults to `#ea3324`.
+   */
+  @Prop()
+  public xArrowColor: Color.Color | string = '#ea3324';
+
+  /**
+   * The color of the translation arrow on the y-axis. Defaults to `#4faf32`.
+   */
+  @Prop()
+  public yArrowColor: Color.Color | string = '#4faf32';
+
+  /**
+   * The color of the translation arrow on the z-axis. Defaults to `#0000ff`.
+   */
+  @Prop()
+  public zArrowColor: Color.Color | string = '#0000ff';
+
+  /**
+   * The color override of the hovered component. Defaults to `#ffff00`.
+   */
+  @Prop()
+  public hoveredColor: Color.Color | string = '#ffff00';
+
+  /**
+   * The color to display when persistence of a transform is pending. Defaults to `#cccccc`.
+   */
+  @Prop()
+  public disabledColor: Color.Color | string = '#cccccc';
+
+  /**
+   * @internal
+   * @private
+   *
+   * Visible for testing.
+   */
+  @Prop({ mutable: true })
+  public hovered?: Mesh;
+
+  private currentPosition?: Vector3.Vector3;
+
+  private widget?: TransformWidget;
+  private dragging?: Mesh;
+  private lastWorldPosition?: Vector3.Vector3;
+
+  private canvasBounds?: DOMRect;
+  private canvasResizeObserver!: ResizeObserver;
+  private canvasRef?: HTMLCanvasElement;
+
+  private hoveredChangeDisposable?: Disposable;
+
+  protected componentDidLoad(): void {
+    window.addEventListener('pointermove', this.handleWindowPointerMove);
```

Comment:
Does this have to be on the window?

Reply:
It likely could be moved to the canvas itself, just with some conditional logic to remove the listener when we're actively transforming. I can take a look at updating that

Reply:
Actually realized as I started looking into this why it was set up on the window to begin with - we need a `pointer-events: none` on the canvas so it doesn't prevent viewer interactions, which means we can't actually listen for the canvas-level events, and need to add to the window